### PR TITLE
ci(flask): support for Flask 1.1 and Python > 3.6

### PR DESCRIPTION
**Pull request by jd**

### CODE REVIEW

**Review by P403n1x87 — APPROVED** [review]

File: tests/contrib/flask/test_request.py [hunk 1]
```diff
@@ -556,9 +555,24 @@ def fivehundred():
...
+            # Flask >= 1.1
+            span_names = [
+                "flask.request",
+                "flask.try_trigger_before_first_request_functions",
+                "flask.preprocess_request",
+                "flask.dispatch_request",
+                "tests.contrib.flask.test_request.fivehundred",
+                "flask.handle_user_exception",
+                "flask.handle_exception",
+                "flask.process_response",
```
Comment (brettlangdon) [inline tests/contrib/flask/test_request.py:569]: sneaky, sneaky


would it read more clear as something like:


```python
            span_names = [
                "flask.request",
                "flask.try_trigger_before_first_request_functions",
                "flask.preprocess_request",
                "flask.dispatch_request",
                "tests.contrib.flask.test_request.fivehundred",
                "flask.handle_user_exception",
            ]
            if FLASK_VERSION_TUPLE >= (1, 1, 0):
                span_names.append("flask.process_response")
            span_names += [
                "flask.do_teardown_request",
                "flask.do_teardown_appcontext",
            ]
```

Not a strong opinion, just thinking if there is a way to make it easier to spot what is different for Flask  >= 1.1.0

File: tox.ini [hunk 2]
```diff
@@ -59,7 +59,7 @@ envlist =
     dogpile_contrib-py{36,37,38,39}-dogpilecache{06,07,08,09,10,}
     falcon_contrib{,_autopatch}-{py27,py35,py36,py37}-falcon{10,11,12,13,14}
     falcon_contrib{,_autopatch}-{py35,py36,py37}-falcon{20}
-    flask_contrib{,_autopatch}-{py27,py35,py36}-flask{010,011,012,10}-blinker-pytest3
+    flask_contrib{,_autopatch}-{py27,py35,py36,py37,py38,py39}-flask{010,011,012,10,11}-blinker-pytest3
```
Comment (brettlangdon) [inline tox.ini:62]: should we also include "latest" here?
```suggestion
    flask_contrib{,_autopatch}-{py27,py35,py36,py37,py38,py39}-flask{010,011,012,10,11,}-blinker-pytest3
```

and then:

```diff
+ flask: flask>=1.1
```

Comment (jd) [reply tox.ini:62]: wfm

File: tests/contrib/flask/test_request.py (hunk 1 above)
Comment (jd) [reply tests/contrib/flask/test_request.py:569]: Sure why not :)

**Review by Kyle-Verhoog — APPROVED** [review]

**Review by brettlangdon — APPROVED** [review]

**Review by Kyle-Verhoog — APPROVED** [review]

**Pull request merged**